src/electionguard-ui/ElectionGuard.UI/ViewModels/ChallengedPopupViewModel.cs

```diff
@@ -54,9 +66,32 @@ private async Task SpoilBallot()
         var answer = await Shell.Current.CurrentPage.DisplayAlert(AppResources.SpoilBallotText, AppResources.SpoilConfirmationText, AppResources.YesText, AppResources.NoText);
         if (answer)
         {
+            // update the ballot data to spoiled
```

Comment:
Completed the converting of the challenged ballot to a spoiled and changing the tally and other structures